examples/audio_embedder/android/app/src/main/java/com/google/mediapipe/examples/audioembedder/AudioEmbedderHelper.kt

```diff
@@ -0,0 +1,177 @@
+        when (currentDelegate) {
+            DELEGATE_CPU -> {
+                baseOptionsBuilder.setDelegate(Delegate.CPU)
+            }
+            DELEGATE_GPU -> {
+                baseOptionsBuilder.setDelegate(Delegate.GPU)
+            }
+        }
+
+        baseOptionsBuilder.setModelAssetPath(MODEL_PATH)
+
+        try {
+            val baseOptions = baseOptionsBuilder.build()
+            val optionsBuilder =
+                AudioEmbedderOptions.builder()
+                    .setBaseOptions(baseOptions)
+                    .setRunningMode(RunningMode.AUDIO_CLIPS)
+            val options = optionsBuilder.build()
+            audioEmbedder = AudioEmbedder.createFromOptions(context, options)
+        } catch (e: IllegalStateException) {
+            listener?.onError(
+                "Audio embedder failed to initialize. See error logs for " +
+                        "details"
+            )
+            Log.e(
+                TAG,
+                "Audio embedder failed to load model with error: " + e.message
+            )
+        } catch (e: RuntimeException) {
+            // This occurs if the model being used does not support GPU
+            listener?.onError(
+                "Audio embedder failed to initialize. See error logs for " +
+                        "details", GPU_ERROR
+            )
+            Log.e(
+                TAG,
+                "Audio embedder failed to load model with error: " + e.message
+            )
+        }
+    }
+
+    //  If both the vectors are aligned, the angle between them
+    //  will be 0. cos 0 = 1. So, mathematically, this distance metric will
+    //  be used to find the most similar audio.
+    fun compare(
+        firstAudioData: AudioData,
+        secondAudioData: AudioData
+    ): ResultBundle? {
+        // Inference time is the difference between the system time at the start and finish of the
+        // process
+        val startTime = SystemClock.uptimeMillis()
+        audioEmbedder?.let {
+            val firstEmbed =
+                it.embed(firstAudioData).embeddingResultList().get().first()
+                    .embeddings()
+                    .first()
+            val secondEmbed =
+                it.embed(secondAudioData).embeddingResultList().get().first()
+                    .embeddings()
+                    .first()
+            val inferenceTimeMs = SystemClock.uptimeMillis() - startTime
+            return ResultBundle(
+                AudioEmbedder.cosineSimilarity(firstEmbed, secondEmbed),
+                inferenceTimeMs
+            )
+        }
+        return null
+    }
+
+    fun clearAudioEmbedder() {
+        audioEmbedder?.close()
+        audioEmbedder = null
+    }
+
+    // Wraps results from inference, the time it takes for inference to be
+    // performed.
+    data class ResultBundle(
+        val similarity: Double,
+        val inferenceTime: Long,
+    )
+
+    companion object {
```

Comment:
ResultBundle and the constants should be at the bottom of the class to fit the conventions of our other samples.